src/auto-evo/AutoEvoExploringTool.cs

```diff
@@ -891,7 +896,16 @@ private void UpdateSpeciesPreview(Species species)
     private void EvolutionaryTreeNodeSelected(int generation, uint id)
     {
         HistoryListMenuIndexChanged(generation);
-        UpdateSpeciesPreview(world.SpeciesHistoryList[generation][id]);
+
+        var species = GetActiveSpeciesDataFromGeneration(generation, id);
+
+        if (species == null)
+        {
+            GD.PrintErr("Couldn't find active species data to show");
```

Comment:
Shouldn't this clear the preview? So that on error the GUI doesn't keep just showing some data that doesn't match what the player clicked.

Reply:
Isn't it possible to set the species to null? Or am I thinking about some other thing.

Reply:
`UpdateSpeciesPreview` relies on species being non-null, so there would be some issues

Reply:
I think you'll need to add a method like `ClearShownSpecies` to `SpeciesDetailsPanelWithFossilisation` to handle this error case in a sensible way.